### ISSUE TITLE
Databricks Dialect: `Set variable x = select y from z` unparsable

### ISSUE THREAD

**Issue by fstg1992** [sqlfluff/sqlfluff#6463]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing of set variable x = select y from z did not parse correctly in databricks dialect.

### Expected Behaviour

Gets parsed.

### Observed Behaviour

Found unparsable section: unparsable:                                               !! Expected: 'Nothing else in FileSegment.'

statement:
[L:  2, P:  1]      |        set_statement:
[L:  2, P:  1]      |            keyword:                                          'SET'
[L:  2, P:  4]      |            whitespace:                                       '     '
[L:  2, P:  9]      |            property_name_identifier:
[L:  2, P:  9]      |                properties_naked_identifier:                  'VARIABLE'
[L:  2, P: 17]      |            whitespace:                                       ' '
[L:  2, P: 18]      |            properties_naked_identifier:                      'process_start_ts'
[L:  2, P: 34]      |    whitespace:                                               ' '
[L:  2, P: 35]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  2, P: 35]      |        equals:                                               '='
[L:  2, P: 36]      |        whitespace:                                           ' '
[L:  2, P: 37]      |        start_bracket:                                        '('
[L:  2, P: 38]      |        word:                                                 'select'
[L:  2, P: 44]      |        whitespace:                                           ' '
[L:  2, P: 45]      |        word:                                                 'a'
[L:  2, P: 46]      |        whitespace:                                           ' '
[L:  2, P: 47]      |        word:                                                 'from'
[L:  2, P: 51]      |        whitespace:                                           ' '
[L:  2, P: 52]      |        word:                                                 'b'
[L:  2, P: 53]      |        end_bracket:                                          ')'
[L:  2, P: 54]      |        semicolon:                                            ';'

### How to reproduce

```sh
echo "set var foo = select bar;" > tst.sql
sqlfluff lint tst.sql --dialect=databricks
```

### Dialect

Databricks

### Version

3.2.5

### Configuration

dialect = "databricks"
templater = "jinja"
max_line_length = 150

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fstg1992** [comment]
It appears, the current databricks implementation (link)[https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-aux-set-variable.html] is identical to the implementation of spark 4.0.0 preview (link)[https://spark.apache.org/docs/4.0.0-preview2/sql-ref-syntax-aux-set-var.html].
would be more sensible to implement fix in spark dialect, probably?

**Comment by Nikhil-24-04** [comment]
SET variable_name = 'value';
